### DISCUSSION TITLE
Quick Question: How to ensure use the old metrics data when re-install the mimr

### DISCUSSION THREAD

**Discussion by Nebulazhang** [grafana/mimir#5992] [Help and support] [answered]
I uninstall the mimir and re-install it again. But I cannot get the old metrics although I use the S3 as the storage.

Is it because I used anonymous as the user

**Comment by pstibrany** [comment, marked as answer]
Hello, did you use a different user before? If you can see the data in S3 bucket, you can move old data under new tenant, and after some time (~30 mins with default config) that old data will be available for querying again.

**Reply by Nebulazhang** [reply]
But I didn't add any tenant, only use the `anonymous` as the tenant. And I found the data in the anonymous dir in my S3 bucket. But I cannot query the older metrics...

**Reply by pstibrany** [reply]
To troubleshoot this, I'd suggest checking bucket-index.json.gz file in S3 under your tenant. It should be updated by compactor every 15 minutes, and contain up-to-date list of the blocks. Queriers and store-gateways then use this file to know which blocks are available for querying.

You can also try to download a block, and list what series are in it (eg. using `tsdb-index` tool from Mimir repository under `tools`), and try to query those series with date range of the block.
